### DISCUSSION TITLE
Should pybricks REPL support f-strings?

### DISCUSSION THREAD

**Discussion by BertLindeman** [pybricks/support#1855] [General]
**Sometimes** I past snippets of my program into the REPL to test things.

I like f-strings as they are (to me) easier to understand than the other formatting syntax.
But in the REPL f-strings are refused in pybricks (at the moment I usie 3.6.0b1)

micropython does support f-strings since 1.12 or 1.13 I think.

The micropython pybricks is based upon micropython 1.20:
```('micropython', (1, 20, 0), 'SPIKE Prime Hub with STM32F413VG', 7430)``` 

Did not want to make an **issue** for this, as I do not know if this is / was a design decision.

And this might not be a good moment to make a change for such a minor thing, now.

Bert

**Comment by laurensvalk** [comment]
Enabling the flag will let us use them on the REPL.

Oddly enough, the question is why they work in scripts when this feature is not enabled. I suppose `mpy-cross` is parsing them on compile time?

**Reply by laurensvalk** [reply]
Enabling f-strings on the REPL takes about 500 bytes on Technic Hub, which is also gradually approaching its maximum firmware size. So my inclination is to leave this disabled since it does work well in scripts. Feel free to make a dedicated issue if you feel we should enable it.

See https://github.com/pybricks/support/issues/1668

**Reply by dlech** [reply]
> I suppose `mpy-cross` is parsing them on compile time?

Yup, `MICROPY_PY_FSTRINGS` is enabled for mpy-cross and the MicroPython implementation of f-strings is to rewrite it to string.format() at compile time.
